app/screens/user_profile/user_profile.js

```diff
@@ -187,6 +216,43 @@ export default class UserProfile extends PureComponent {
         };
     };
 
+    goToEditProfile = () => {
+        const {user: currentUser} = this.props;
+        const {formatMessage} = this.context.intl;
+
+        const {navigator, theme} = this.props;
+        const options = {
+            screen: 'EditProfile',
+            title: formatMessage({id: 'mobile.routes.edit_profile', defaultMessage: 'Edit Profile'}),
+            animated: true,
+            backButtonTitle: '',
+            passProps: {currentUser},
+            navigatorStyle: {
+                navBarTextColor: theme.sidebarHeaderTextColor,
+                navBarBackgroundColor: theme.sidebarHeaderBg,
+                navBarButtonColor: theme.sidebarHeaderTextColor,
+                screenBackgroundColor: theme.centerChannelBg,
+            },
+        };
+
+        requestAnimationFrame(() => {
+            navigator.push(options);
+        });
+    };
+
+    onNavigatorEvent = (event) => {
+        if (event.type === 'NavBarButtonPress') {
+            switch (event.id) {
+            case this.rightButton.id:
+                this.goToEditProfile();
+                break;
+            case 'close-settings':
+                this.close();
+                break;
+            }
+        }
+    };
```

Comment:
Please add unit test for `goToEditProfile` and `onNavigatorEvent`.

Reply:
@saturninoabril Thanks for the review! Just committed unit tests for user_profile.
I'm not familiar unit tests yet, please check the code for any errors!